app/models/molecule.rb

```diff
@@ -246,6 +245,20 @@ def unique_molecule_name(new_name)
     !mns.include?(new_name)
   end
 
+  def self.svg_reprocess(svg, molfile)
+    return svg unless Rails.configuration.try(:ketcher_service).try(:url).present?
+    return svg if svg.present? && !svg&.include?('Open Babel')
+
+    svg = KetcherService::RenderSvg.svg(molfile)
+
+    if svg&.present?
+      svg = Ketcherails::SVGProcessor.new(svg)
+      svg.centered_and_scaled_svg
+    else
+      Chemotion::OpenBabelService.svg_from_molfile(molfile)
```

Comment:
line unreachable if any error is raised by KetcherService 
eg: when the service is configured but down, then it is not possible to create sample with smiles
(see comment below)